core/arch/riscv64/mangle.c

```diff
@@ -1021,8 +1044,8 @@ mangle_exclusive_store(dcontext_t *dcontext, instrlist_t *ilist, instr_t *instr,
     dst1 = instr_get_dst(instr, 1);
     ASSERT(opnd_is_base_disp(dst0));
     opsz = instr_get_opcode(instr) == OP_sc_d ? OPSZ_8 : OPSZ_4;
-    scratch_reg1 = pick_scratch_reg(dcontext, instr, DR_REG_NULL, &slot1);
-    scratch_reg2 = pick_scratch_reg(dcontext, instr, scratch_reg1, &slot2);
+    scratch_reg1 = pick_scratch_reg(dcontext, instr, DR_REG_NULL, DR_REG_NULL, &slot1);
+    scratch_reg2 = pick_scratch_reg(dcontext, instr, scratch_reg1, DR_REG_NULL, &slot2);
 
     /* Spill scratch registers. */
     PRE(ilist, instr, instr_create_save_to_tls(dcontext, scratch_reg1, slot1));
```

Comment:
Explain how this was tested in the PR description.  I see it says an automated test is forthcoming but there must have been manual testing of this change: explain how that was done, to give confidence that this code works.

Reply:
Okay, the description has been updated with a short description of testing.